course5/homework11.2/homework11.2.cc

```diff
@@ -0,0 +1,57 @@
+#include <benchmark/benchmark.h>
+#include <vector>
+#include <iostream>
+
+static void BM_VectorSequential(benchmark::State & state)
+{
+	std::vector<int> v(10000000, 42);
+	while (state.KeepRunning())
+	{
+		int result = 0;
+		for (size_t i = 0; i < v.size(); i++)
+		{
+			size_t index;
+			benchmark::DoNotOptimize(index = i % 2 ? i : i - 0);
```

Comment:
This doesn't really prevent the optimisation of the expression inside. The function only stops the compiler from throwing the expression away.